server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -538,6 +538,9 @@ public void doRun()
                 cleanShutdown = false;
                 abandonSegment(truncatedTime, sink);
               }
+            } finally {
+              metrics.incrementPersistTimeMillis(mergeStopwatch.elapsed(TimeUnit.MILLISECONDS));
```

Comment:
this also includes segment publish time, it might be worthwhile to only count actual merge time if handoff can take a while?